release/create_candidate_pr.py

```diff
@@ -0,0 +1,79 @@
+# pylint: disable=redefined-outer-name, invalid-name
+
+"""
+Creates a new release candidate branch and makes a pull request
+for it into the 'release' branch.
+"""
+
+from __future__ import print_function
+
+import argparse
+
+from release.get_token import get_token
+from release.github_api import GithubApi, RequestFailed
+from release import utils
+
+
+def _build_parser():
+    """ Creates an option parser for this command """
+    expected_date = utils.default_expected_release_date()
+
+    result = argparse.ArgumentParser(
+        description="""
+        Create a new release candidate branch and associated pull request.
+        """
+    )
+
+    result.add_argument(
+        '--release-date', nargs=1, default=expected_date,
+        help="""
+        Specify a date that the release branch is expected to be deployed.
+        Should be in YYYY-MM-DD format. If not passed, defaults to the
+        next upcoming Tuesday, which is currently {date}.
+        """.format(date=expected_date.date().isoformat()),
+        )
+    result.add_argument(
+        '--org', nargs=1, default='edx',
+        help="""
+        Specify a github organization to work under. Default is 'edx'.
+        """)
+    result.add_argument(
+        '--repo', nargs=1, default='edx-platform',
+        help="""
+        Specify a github repository to work with. Default is 'edx-platform'.
+        """)
+
+    return result
+
+parser = _build_parser()
+args = parser.parse_args()
+
+token = get_token()
+github_api = GithubApi(args.org, args.repo, token)
+
+print("Fetching commits...")
+try:
+    commit = utils.most_recent_good_commit(github_api)
+    commit_hash = commit['sha']
+    commit_message = commit['commit']['message']
+    message = utils.extract_message_summary(commit_message)
+except utils.NoValidCommitsError:
+    print("Couldn't find a recent commit without test failures. Aborting")
+
+branch_name = utils.rc_branch_name_for_date(args.release_date.date())
+
+print("Branching {rc} off {sha}. ({msg})"
+      .format(rc=branch_name, sha=commit_hash, msg=message))
+try:
+    github_api.create_branch(branch_name, commit_hash)
+except RequestFailed as e:
+    print("Unable to create branch. Aborting")
+    raise e
+
+print("Creating Pull Request for {rc} into master".format(rc=branch_name))
```

Comment:
I just noticed this. The rc PR needs to go against the release branch. Also, we should make sure that the nefarious release tag is not an issue when creating the PR against the branch (if we need to remove it when making the PR, that is fine).